compair/static/modules/group/group-module.js

```diff
@@ -61,23 +61,39 @@ module.factory(
 /***** Controllers *****/
 module.controller(
     'AddGroupModalController',
-    ["$rootScope", "$scope", "$uibModalInstance", "GroupResource",
-    function ($rootScope, $scope, $uibModalInstance, GroupResource) {
+    ["$rootScope", "$scope", "$uibModalInstance", "$filter", "Toaster", "GroupResource",
+    function ($rootScope, $scope, $uibModalInstance, $filter, Toaster, GroupResource) {
         $scope.group = {};
         $scope.modalInstance = $uibModalInstance;
         $scope.submitted = false;
 
         $scope.groupSubmit = function () {
             $scope.submitted = true;
 
-            GroupResource.save({'courseId': $scope.courseId}, $scope.group).$promise.then(
-                function (ret) {
-                    $scope.group = ret;
-                    $uibModalInstance.close($scope.group.id);
+            // Ensure name is unique
+            $scope.groups = GroupResource.get({'courseId': $scope.courseId}).$promise.then(
```

Comment:
Why not allow the backend to handle it? (it is generally good to do it in the front end but since it requires an api call, I think its better to do it in the backend the one time)

Reply:
Just because we can provide a more friendly message and be specific about the action that needs to be taken here (i.e. "Please enter another group name and press Save.") thoughts?